File: angular/src/app/interfaces/order.ts
import { AggregatedProduct, Product } from './product';

export interface Order {
  id?: string;
  status: 'PENDING' | 'CANCELLED' | 'CONFIRMED';
  buyerId: string;
  sellerId: string;
  product: Product;
  orderPlacedAt?: Date;
  quantity: number;
}

export interface PersonalOrder {
  pending: Order[];
  confirmed: Order[];
  cancelled: Order[];
}

export interface CartItem {
  id?: string;
  status: 'PENDING' | 'CANCELLED' | 'CONFIRMED';
  buyerId: string;
  sellerId: string;
  product: Product;
  addedToCartAt?: Date;
  quantity: number;
}

export interface CartResponse {
  cart: Cart;
  processed: boolean;
  orderModifications: OrderModifications;
}

export interface OrderModifications {
  notes: Set<string>;
  modifications: Order[];
}

export interface OrderStatusUpdate {
  id: string;
  status: 'CANCELLED' | 'CONFIRMED';
}

export interface Cart {
  orders: CartItem[];
}

export type DashboardItem = Order | AggregatedProduct;

File: product-service/src/main/java/com/gritlab/buy01/productservice/model/Order.java
package com.gritlab.buy01.productservice.model;

import java.io.Serializable;
import java.util.Date;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.gritlab.buy01.productservice.model.enums.OrderStatus;

import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Document(collection = "orders")
@Data
public class Order implements Serializable {
  @Id private String id;

  private String sellerId;

  private String buyerId;

  private ProductDTO product;

  private Integer quantity;

  private OrderStatus status;

  private Date orderPlacedAt;

  public Order(String sellerId, String buyerId, ProductDTO product, Integer quantity) {
    this.sellerId = sellerId;
    this.buyerId = buyerId;
    this.product = product;
    this.quantity = quantity;

    this.status = OrderStatus.PENDING;
  }
}
